news-bundle/src/Controller/NewsFeedController.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\NewsBundle\Controller;
+
+use Contao\CoreBundle\Asset\ContaoContext;
+use Contao\CoreBundle\Controller\AbstractController;
+use Contao\CoreBundle\Routing\Page\DynamicRouteInterface;
+use Contao\CoreBundle\Routing\Page\PageRoute;
+use Contao\CoreBundle\ServiceAnnotation\Page;
+use Contao\NewsBundle\Event\FetchArticlesForFeedEvent;
+use Contao\NewsBundle\Event\TransformArticleForFeedEvent;
+use Contao\PageModel;
+use Contao\StringUtil;
+use FeedIo\Feed;
+use FeedIo\Specification;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * @Page(contentComposition=false)
+ */
+class NewsFeedController extends AbstractController implements DynamicRouteInterface
+{
+    public const TYPE = 'news_feed';
+
+    // TODO: Fix content types, maybe make it configurable in the page settings
+    public static $contentTypes = [
+        'atom' => 'application/atom+xml',
+        'json' => 'application/feed+json',
+        'rss' => 'application/rss+xml',
+    ];
+
+    private $urlSuffixes = [
+        'atom' => '.xml',
+        'json' => '.json',
+        'rss' => '.xml',
+    ];
+
+    public function __construct(private readonly ContaoContext $contaoContext, private readonly Specification $specification)
+    {
+    }
+
+    public function __invoke(Request $request, PageModel $pageModel): Response
+    {
+        $this->initializeContaoFramework();
+
+        $staticUrl = $this->contaoContext->getStaticUrl();
+        $baseUrl = $staticUrl ?: $request->getSchemeAndHttpHost();
+
+        $feed = (new Feed())
+            ->setTitle($pageModel->title)
+            ->setDescription($pageModel->description)
+            ->setLanguage($pageModel->language)
+        ;
+
+        $event = new FetchArticlesForFeedEvent($feed, $request, $pageModel);
+        $this->container->get('event_dispatcher')->dispatch($event, FetchArticlesForFeedEvent::NAME);
```

Comment:
If the event has only one purpose, I don't think it needs a name at all, since Symfony can register it on the class name (applies to the `TransformArticleForFeedEvent` as well)